sentry.go

```diff
@@ -64,56 +70,60 @@ func (s sentryHook) Levels() []logrus.Level {
 }
 
 func (s sentryHook) Fire(e *logrus.Entry) error {
-	if s.c == nil {
-		// raven.Client works when it is nil, but skip the useless work and don't hang on Fatal
+	if sentry.CurrentHub().Client() == nil {
 		return nil
 	}
 
-	p := raven.Packet{
-		ServerName: s.hostname,
-		Interfaces: []raven.Interface{
-			// ignore the stack frames for the Fire function itself
-			// the logrus machinery that invoked Fire will also be hidden
-			// because it is not an "in-app" library
-			raven.NewStacktrace(2, 3, []string{"main", "github.com/stripe/veneur"}),
+	event := sentry.NewEvent()
+	event.ServerName = s.hostname
+
+	stacktrace := sentry.NewStacktrace()
+	if len(stacktrace.Frames) >= 2 {
+		// Very carefully, filter out the frame for ConsumePanic itself,
+		// and the frame for the deferred function that invoked
+		// ConsumePanic.
+		stacktrace.Frames = stacktrace.Frames[:len(stacktrace.Frames)-2]
+	}
+	event.Exception = []sentry.Exception{
+		sentry.Exception{
+			Stacktrace: stacktrace,
 		},
 	}
 
 	packetExtraLength := len(e.Data)
 	if err, ok := e.Data[logrus.ErrorKey].(error); ok {
-		p.Message = err.Error()
+		event.Message = err.Error()
 		// don't send the error as an extra field
 		packetExtraLength--
 	} else {
-		p.Message = e.Message
+		event.Message = e.Message
 	}
 
-	p.Extra = make(map[string]interface{}, packetExtraLength)
+	event.Extra = make(map[string]interface{}, packetExtraLength)
 	for k, v := range e.Data {
 		if k == logrus.ErrorKey {
 			continue // already handled this key, don't put it into the Extra hash
 		}
-		p.Extra[k] = v
+		event.Extra[k] = v
 	}
 
 	switch e.Level {
 	case logrus.FatalLevel, logrus.PanicLevel:
-		p.Level = raven.FATAL
+		event.Level = sentry.LevelFatal
 	case logrus.ErrorLevel:
-		p.Level = raven.ERROR
+		event.Level = sentry.LevelError
 	case logrus.WarnLevel:
-		p.Level = raven.WARNING
+		event.Level = sentry.LevelWarning
 	case logrus.InfoLevel:
-		p.Level = raven.INFO
+		event.Level = sentry.LevelInfo
 	case logrus.DebugLevel:
-		p.Level = raven.DEBUG
+		event.Level = sentry.LevelDebug
 	}
 
-	_, ch := s.c.Capture(&p, nil)
-
+	sentry.CaptureEvent(event)
 	if e.Level == logrus.PanicLevel || e.Level == logrus.FatalLevel {
-		// we don't want the program to terminate before reporting to sentry
-		return <-ch
+		// TODO: what to do when timed out?
+		sentry.Flush(10 * time.Second)
```

Comment:
We should abstract this timeout somewhere at the top-level (just so it's easy to check)